### ISSUE TITLE
Sample aggregation 'average' produces invalid results

### ISSUE THREAD

**Issue by rdaforno** [ETHZ-TEC/RocketLogger#42]
When using sample aggregation, the result of the analog channels is invalid:
```
rocketlogger start --rate=10 --samples=10 --channel=V1,V2 --format=csv --aggregate=average
```
Output (both voltages are incorrect here):
```
,DI1,DI2,DI3,DI4,DI5,DI6,V1 [10nV],V2 [10nV]
1634738383.830396219,0,0,0,0,0,0,34692658,42944112
,0,0,0,0,0,0,34699521,42944214
,0,0,0,0,0,0,34702530,42944052
,0,0,0,0,0,0,34703154,42944006
,0,0,0,0,0,0,34705673,42944167
,0,0,0,0,0,0,34704689,42944109
,0,0,0,0,0,0,34702425,42944131
,0,0,0,0,0,0,34701461,42944191
,0,0,0,0,0,0,34698240,42944049
,0,0,0,0,0,0,34695438,42944172
```
Without the 'average' aggregation, the voltages are correct:
```
rocketlogger start --rate=10 --samples=10 --channel=V1,V2 --format=csv
```
Output:
```
,DI1,DI2,DI3,DI4,DI5,DI6,V1 [10nV],V2 [10nV]
1634738613.673547427,0,0,0,0,0,0,335338378,-6359
,0,0,0,0,0,0,335343511,-4524
,0,0,0,0,0,0,335351820,-4891
,0,0,0,0,0,0,335348765,-5747
,0,0,0,0,0,0,335350354,-5136
,0,0,0,0,0,0,335351087,-5136
,0,0,0,0,0,0,335351942,-4524
,0,0,0,0,0,0,335351453,-4280
,0,0,0,0,0,0,335353408,-4891
,0,0,0,0,0,0,335353408,-6603
```

This happens due to an overflow (32-bit signed).

**Comment by lsigrist** [comment]
Thanks @rdaforno for reporting and narrowing down the root cause!

* confirmed for both versions 2.0.0 (reproduced) and 1.1.6 (code inspection)
* data pipeline analysis leads to conclusion that `--aggregate=average` may result in corrupted values for any supported sample rate below 1 kSps

**Comment by lsigrist** [comment]
fix verified using 0c130ac
